react/states_and_effects/how_to_deal_with_side_effects.md

```diff
@@ -0,0 +1,236 @@
+### Introduction
+
+Certain components in react need to be interacting with other things outside the component itself. These other things can be anything from querying data from a server to synchronizing with the position of the component on the webpage or even sending some data to a server when necessary. This leads to the arising of a new concept called 'effects'. 
+
+While we are already familiar with rendering code and adding event handlers, it is not enough for all uses, like when you want to connect to your server and fetch messages to show to a user. Effects let you run some code to synchronize your component as necessary, on rendering or a reactive/state value change rather than on a particular event.
+
+Similar to how we have the `useState` hook, React offers us a handy `useEffect` hook to use effects in our components.
+
+### Lesson Overview
+
+This section contains a general overview of topics that you will learn in this lesson.
+
+*   What are effects?
+*   How are effects used in React?
+*   What are the different parts of an `useEffect` hook?
+*   When should I use an effect?
+
+### Using Effect Saves The Day
```

Comment:
We could add three fourth level headings here to make the section more digestible

```
#### The useEffect hook

#### The dependency array

#### The cleanup function

```